wgpu-core/src/lib.rs

```diff
@@ -254,7 +254,7 @@ macro_rules! gfx_select {
         // Note: For some reason the cfg aliases defined in build.rs don't succesfully apply in this
         // macro so we must specify their equivalents manually
         match $id.backend() {
-            #[cfg(all(not(target_arch = "wasm32"), any(not(any(target_os = "ios", target_os = "macos")), feature = "gfx-backend-vulkan")))]
+            #[cfg(all(not(target_arch = "wasm32"), any(not(any(target_os = "ios", target_os = "macos")), feature = "vulkan-portability")))]
```

Comment:
let's leave a comment saying why the check if for a feature that this crate doesn't have

Reply:
Is 
```
"This macro gets mostly used in the `wgpu-rs` crate, which defines the `vulkan-portability` feature, thus the need for a check to it here"
```
sufficient? I want approval before committing so I don't commit many times for a simple change.

Reply:
Come to think of it again, we shouldn't go this route at all. wgpu-core doesn't have this feature, so adding checks for it makes it a very pool abstraction. Instead, it should be written in a way that self-contained within wgpu-core.
I think there is such a way: instead of checking for the feature inside the macro, how about we try checking it outside?
```rust
#[cfg(feature = "gfx-backend-vulkan")]
#[macro_export]
macro_rules! gfx_select {
}
#[cfg(not(feature = "gfx-backend-vulkan"))]
#[macro_export]
macro_rules! gfx_select {
}
```
Would you want to try out this approach?

Reply:
Sure, I can try that.
If I understand correctly, in the case that `gfx-backend-vulkan` isn't enabled, should the Vulkan backend not be a option at all?
And if it is enabled, should I assume only Vulkan or GL backend depending on wasm to be the architecture or not?

Reply:
> If I understand correctly, in the case that gfx-backend-vulkan isn't enabled, should the Vulkan backend not be a option at all?

My pseudo-code was obviously incomplete.
We need to double check how `feature = xxx` works if `xxx` is non-optional dependency.
The Vulkan backend is always either unconditionally enabled, or behind a feature.

> should I assume only Vulkan or GL backend depending on wasm to be the architecture or not?

Only GL backend makes sense for WASM here

Reply:
I didn't have time to work on this with uni exams and work, but when I can, is it still beneficial for me to work on this since you removed support for it on Apple Platforms?

Reply:
It depends on whether you need those Vulkan Validation layers, or just looking for things to help us with.
if you are just looking to help, then no need to resume this work. There are plenty of issues open!